Trib 104

Jakersnell opened the pull request:
trib-104

[X] Create handler for delete button
[X] Create delete method
[X] Create delete method
[X] Redirect back to list view upon confirm deletion
[X] Update model upon deletion
[X] Get id param from route
[X] Add id param to url in router
[X] Pass id param to url in template
[X] Add test for page
[X] Disable removal of connection when user is non requesting user

Jakersnell committed:
Add 'ConnectApiService.removeConnection' method.

Jakersnell committed:
Add method 'ConnectModelService.removeConnection'.

Jakersnell committed:
Remove trailing '/'

Jakersnell committed:
rename userId to more explanatory name, change type to number

Jakersnell committed:
remove comment

Jakersnell committed:
Use dot operator instead of bracket operator for type User

Jakersnell committed:
Initialize 'ConnectionDetailsPage.initConnectedWithUserId'

Jakersnell committed:
Add handler in button to remove connection upon confirm click

Jakersnell committed:
Add url parameter :userId to Conn

Jakersnell committed:
Add url parameter 'connectedWithUserId' to url for 'ConnectionDetailsPage'.

Jakersnell committed:
Change 'userId' to 'connectedWithUserId' in paramMap.

Jakersnell committed:
Navigate to 'ListConnectionsPage' upon removing connection.

Jakersnell committed:
Configure url to route to details page.
modified:   src/app/pages/connect-page/list-connections/list-connections.page.html

Jakersnell committed:
Route back to correct url upon removing connection.
modified:   src/app/pages/connect-page/connection-details/connection-details.page.ts

Jakersnell committed:
Create test file for connection details page.
new file:   src/app/pages/connect-page/connection-details/connection-details.spec.ts

Jakersnell committed:
Add test file for connection-details.page.ts
 Changes to be committed:
	modified:   src/app/pages/connect-page/connection-details/connection-details.spec.ts

Jakersnell committed:
disable removal of connection when user is not requesting user

haxwell committed:
Merge branch 'develop' into trib-104

Jakersnell committed:
Merge branch 'develop' into trib-104

Jakersnell committed:
Merge branch 'trib-104' of https://github.com/Jakersnell/tribe-app-frontend into trib-104

Jakersnell committed:
Merge branch 'develop' into trib-104

Jakersnell committed:
update tests for passing 'shouldCreate'

haxwell commented on src/app/pages/connect-page/connection-details/connection-details.page.ts:
```diff
@@ -38,7 +64,11 @@ export class ConnectionDetailsPage implements OnInit{
+          text: 'Yes!',
+          handler: () => {
+            this._connectModelService.removeConnection(this.connectedWithUserId);
+            this._router.navigate(['/connect/list-connections']);
```
There is a promise being returned from the model service call. Use it, so that you call the router until the model service call is done done.

haxwell commented on src/app/pages/connect-page/_services/connect.model.service.ts:
```diff
@@ -30,11 +30,25 @@ export class ConnectModelService {
     }
+
+    removeConnection(connectedWithUserId: number) {
+        return new Promise((resolve, reject) => {
```
No need to return a new promise here, you can return the one from the apiService call. Likewise, above in your init() call..

haxwell commented on src/app/pages/connect-page/_services/connect.api.service.ts:
```diff
@@ -40,4 +40,23 @@ export class ConnectApiService {
+          connectedWithUserId: connectedWithUserId
+        };
+        return new Promise((resolve, reject) => {
+            this._apiService.delete(url, connectionRemovalRequest).subscribe(
```
apiService.delete() already returns a promise.. you can just return that.. no need to wrap it.

* Side thought.. we are probably doing this in a bunch of places, unnecessarily wrapping a promise.. we should fix that.

Jakersnell committed:
Navigate with router upon promise resolving.

Jakersnell committed:
Tap into existing promise instead of creating new promise.
	modified:   src/app/pages/connect-page/_services/connect.model.service.ts

Jakersnell committed:
Add e2e tests for connection details page.
	modified:   cypress/e2e/connect.page.cy.js

Jakersnell committed:
Add 'data-test' attributes for better accessibility and more flexibility during e2e tests.
	modified:   src/app/pages/connect-page/connection-details/connection-details.page.html

Jakersnell committed:
Route to list-connections upon delete method's promise resolving.
	modified:   src/app/pages/connect-page/connection-details/connection-details.page.ts

Jakersnell committed:
Add 'role' attribute to alert service button for better acessibility and flexibility during e2e testing.
	modified:   src/app/pages/connect-page/connection-details/connection-details.page.ts

Jakersnell committed:
Structure connections mock for easy resetting.
	modified:   cypress/e2e/connect.page.cy.js
